applications/serial_lte_modem/src/slm_at_tcp_proxy.c

```diff
@@ -435,76 +452,199 @@ static int tcp_data_save(uint8_t *data, uint32_t length)
 	return ring_buf_put(&data_buf, data, length);
 }
 
-static void tcp_thread_func(void *p1, void *p2, void *p3)
+int tcpsvr_input(int infd)
 {
 	int ret;
-	struct pollfd fds;
-	int sock;
-
-	ARG_UNUSED(p1);
-	ARG_UNUSED(p2);
-	ARG_UNUSED(p3);
 
-thread_entry:
-	if (proxy.role == AT_TCP_ROLE_SERVER) {
-		socklen_t len = sizeof(struct sockaddr_in);
+	if (fds[infd].fd == proxy.sock) {
+		socklen_t len;
 		char peer_addr[INET_ADDRSTRLEN];
 
+		len = sizeof(struct sockaddr_in);
+		if (nfds >= MAX_POLL_FD) {
+			LOG_WRN("Full. Can not accept"
+			" connection.");
+			return -ECONNREFUSED;
```

Comment:
I think we should rather `accept()` anyway and then `close()` the socket rightaway. Otherwise those pending connections will pile up in the modem (and I guess we could run out of mem if there are many).

Reply:
True. The pending unwanted connection is now accepted and closed.